### PR TITLE
K8SPG-266: Collect telemetry

### PR DESCRIPTION
**Pull request by egegunes**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/c13f19c2/init-deploy.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/c13f19c2/demand-backup.log) |
| start-from-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/c13f19c2/start-from-backup.log) |
| We run 3 out of 3|

commit: https://github.com/percona/percona-postgresql-operator/pull/314/commits/c13f19c23c3a6b83dc8080fe6f641a434ba7d527
image: `perconalab/percona-postgresql-operator:PR-314-c13f19c2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/c13f19c2/init-deploy.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/c13f19c2/demand-backup.log) |
| start-from-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/c13f19c2/start-from-backup.log) |
| We run 3 out of 3|

commit: https://github.com/percona/percona-postgresql-operator/pull/314/commits/c13f19c23c3a6b83dc8080fe6f641a434ba7d527
image: `perconalab/percona-postgresql-operator:PR-314-c13f19c2`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/e784664d/init-deploy.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/e784664d/demand-backup.log) |
| start-from-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/e784664d/start-from-backup.log) |
| We run 3 out of 3|

commit: https://github.com/percona/percona-postgresql-operator/pull/314/commits/e784664d7681366aba8d0463fc33b9520131e4ec
image: `perconalab/percona-postgresql-operator:PR-314-e784664d`

File: percona/version/version.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"github.com/pkg/errors"
+
+	"github.com/percona/percona-postgresql-operator/internal/logging"
+	"github.com/percona/percona-postgresql-operator/percona/version/service/client"
+	"github.com/percona/percona-postgresql-operator/percona/version/service/client/version_service"
+	"github.com/percona/percona-postgresql-operator/pkg/apis/pg.percona.com/v2beta1"
+)
+
+const (
+	Version     = "2.0.0"
```
Comment (hors) [inline percona/version/version.go:20]: Do we need it? We will need to change it every release.

File: percona/version/version.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	if err != nil {
+		// we don't return here to not block execution just because we can't phone home
+		log.Error(err, fmt.Sprintf("failed to send telemetry to %s", v2beta1.GetDefaultVersionServiceEndpoint()))
+	}
+
+	return nil
+}
+
+func telemetryEnabled() bool {
+	value, ok := os.LookupEnv("DISABLE_TELEMETRY")
```
Comment (hors) [inline percona/version/version.go:54]: please add it to operator/bundle yaml

**Review by hors — CHANGES_REQUESTED** [review]
We need to have a test. We have it for all our operators.

File: cmd/postgres-operator/main.go [hunk 3]
```diff
@@ -190,3 +190,102 @@ func isOpenshift(ctx context.Context, cfg *rest.Config) bool {
...
+func isGKE(ctx context.Context, cfg *rest.Config) bool {
+	log := logging.FromContext(ctx)
+
+	const groupName, kind = "cloud.google.com", "BackendConfig"
+
+	client, err := discovery.NewDiscoveryClientForConfig(cfg)
+	assertNoError(err)
+
+	groups, err := client.ServerGroups()
+	if err != nil {
```
Comment (inelpandzic) [inline cmd/postgres-operator/main.go:203]: You don't need `if err != nil` checks, just use `assertNoError(err)`. Applies to other places as well.

File: cmd/postgres-operator/main.go [hunk 4]
```diff
@@ -190,3 +190,102 @@ func isOpenshift(ctx context.Context, cfg *rest.Config) bool {
...
+}
+
+// getServerVersion returns the stringified server version (i.e., the same info `kubectl version`
+// returns for the server)
+// Any errors encountered will be logged and will return an empty string
+func getServerVersion(ctx context.Context, cfg *rest.Config) string {
+	log := logging.FromContext(ctx)
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+	if err != nil {
+		log.V(1).Info("upgrade check issue: could not retrieve discovery client",
```
Comment (inelpandzic) [inline cmd/postgres-operator/main.go:280]: Maybe have `Error` log here? Applies to other places as well.

**Review by inelpandzic — APPROVED** [review]
Besides a couple of comments, this looks good to me.

**Comment by cap1984** [comment]
Please add `DISABLE_TELEMPTRY` just like at other operators have

File: percona/version/version.go (hunk 2 above)
Comment (egegunes) [reply percona/version/version.go:54]: https://github.com/percona/percona-postgresql-operator/pull/314/commits/d1069a85b8a83139a89ac1b830e6a9e1a201298b

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/08bc8078/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/08bc8078/demand-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/08bc8078/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-314/08bc8078/monitoring.log) |
| We run 4 out of 4|

commit: https://github.com/percona/percona-postgresql-operator/pull/314/commits/08bc8078402234f5287d26572aaa533964348bec
image: `perconalab/percona-postgresql-operator:PR-314-08bc8078`

**Review by hors — APPROVED** [review]

**Pull request merged**